Modules/Common/StellarLikeMainModule.php

```diff
@@ -0,0 +1,985 @@
+                                    $this->parse_account_credited($op, $tx_is_failed, $effect, $sort_key, $events, $currencies_to_process);
+                                    break;
+                                }
+                                case 'claimable_balance_clawed_back':
+                                case 'claimable_balance_sponsorship_removed':
+                                    break;
+                                default:
+                                    throw new ModuleError("Unknown operation type: " . $effect['type'] . " operation: " . $op['id']);
+                            }
+                        }
+                        break;
+                    }
+                // if operation doesn't have any monetary operations -- we skip it, only fees we pay in Main Module
+                case 'create_passive_sell_offer': // 210453732511657987
+                case 'change_trust':
+                case 'manage_buy_offer':
+                case 'manage_sell_offer':
+                case 'set_options': // 210027804900413445
+                case 'allow_trust': // 210027804899786755
+                case 'begin_sponsoring_future_reserves':    // 210027804900413441
+                case 'set_trust_line_flags':        // 210027774835036161
+                case 'end_sponsoring_future_reserves': // 210457202845294614
+                case 'bump_sequence': // 210457151305424897
+                case 'revoke_sponsorship': //210965907361755138
+                    break;
+                default:
+                    throw new ModuleError("Unknown operation type: " . $op['type'] . " operation: " . $op['id']);
+            }
+        }
+
+        $currencies_to_process = array_values(array_unique($currencies_to_process)); // Removing duplicates
+        $currencies_to_process = check_existing_currencies($currencies_to_process, $this->currency_format); // Removes already known currencies
+
+        foreach ($currencies_to_process as $currency) 
+        {
+            if ($currency !== null)
+            {
+                $currencies[] = [
+                    'id'       => $currency,
+                    'name'     => explode(':', $currency)[1],
+                    'symbol'   => explode(':', $currency)[0],
+                    'decimals' => 7,
+                ];
+            }
+        }
+
+        ////////////////
+        // Processing //
+        ////////////////
+
+        foreach ($events as &$event)
+        {
+            $event['block'] = $block_id;
+            $event['time'] = $this->block_time;
+
+            if (is_null($event['currency']))
+                $event['currency'] = $this->native_currency;
+        }
+
+        $this->set_return_events($events);
+        $this->set_return_currencies($currencies);
+    }
+
+    // Getting balances from the node
+    // wtf what to do with it?
+    final public function api_get_balance(string $address, array $currencies): array
```

Comment:
@Har01d Due to the some PHP features we can't override two functions with the same name in one module without calling the rain